# <center>Being Invisible! </center>

## Program to click a picture of yourself:

In [86]:
import numpy as np
import cv2
import time

In [87]:
# Initialize a video capture object
cap = cv2.VideoCapture(0)

#Read a single image frome
_, frame = cap.read()

# Display the image frame, with window titled 'Your Picture'
cv2.imshow('Your Picture', frame)

# Wait for 2 seconds or until a key is pressed
cv2.waitKey(2000)

# Destroy all windows
cv2.destroyAllWindows()

# Release the camera
cap.release()

Thus the above code captures an image frame which is diplayed for two seconds. To save this image use:

In [88]:
cv2.imwrite(filename = 'yourpicture.png', img = frame)

True

## Program to take a live video:

In [89]:
import numpy as np
import cv2
import time

In [90]:
# Initialize a video capture object
video = cv2.VideoCapture(0)

while(True):
    #Read an image frome
    _, frame = video.read()
    
    # Display the image frame, with window titled 'Your Picture'
    cv2.imshow('Your Video', frame)

    # Wait for 1 milliseconds or until a key is pressed
    key = cv2.waitKey(1) & 0xFF 
    
    # Exit the loop when q is pressed
    if key == ord('q'):
        break
        
# Destroy all windows
cv2.destroyAllWindows()

# Release the camera
video.release()

Thus we have captured a live feed video.

**Note :** We use the key = cv2.waitKey(1) & 0xFF because we are running a 64bit machine. See [Stackoverflow](https://stackoverflow.com/questions/35372700/whats-0xff-for-in-cv2-waitkey1)

## Program to Crop your image:

In [91]:
import numpy as np
import cv2
import time

In [92]:
# Initialize a video capture object
cap = cv2.VideoCapture(0)

#Read a single image frome
_, frame = cap.read()

# Display the image frame, with window titled 'Your Picture'
cv2.imshow('Your Picture', frame)

# Wait for 2 seconds or until a key is pressed
cv2.waitKey(2000)

# Destroy all windows
cv2.destroyAllWindows()

# Release the camera
cap.release()

In [93]:
# The shape of our image is:
frame.shape

(480, 640, 3)

**Note :** In OpenCV the dimensions are given as: (height, width, channels)

The original image is:

In [94]:
cv2.imshow('Original', frame)

cv2.waitKey(0)

cv2.destroyAllWindows()

Assuming you are sitting in the middle, and occupy 100 pixels vertically & horizontally, the ROI will be [140:340, 220:420]

In [95]:
crop = frame[140:340, 220:420]

In [96]:
cv2.imshow('Cropped', crop)

cv2.waitKey(0)

cv2.destroyAllWindows()

After seeing the cropped image, it is clear that we should extend the vertical limits till 480 and also increase the horizonatal limits as well. After some trial and error we get the following limits to crop your picture from the image frame:

In [97]:
crop = frame[140:420, 170:510]

cv2.imshow('Cropped', crop)

cv2.waitKey(0)

cv2.destroyAllWindows()

In [98]:
crop.shape

(280, 340, 3)

Thus we have succesfully cropped our picture out of the image frame, the cropped image can easily be saved using the cv2.imwrite() function as shown above

## References Used:

- https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_tutorials.html
- https://stackoverflow.com/questions/35372700/whats-0xff-for-in-cv2-waitkey1
- https://stackoverflow.com/questions/15589517/how-to-crop-an-image-in-opencv-using-python
- https://www.youtube.com/watch?v=-ZrDjwXZGxI